scripts/auto-rebase/presubmit.py

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import re
+import json
+import fnmatch
+from enum import Enum
+from pathlib import Path
+
+REBASE_SCRIPT = "./scripts/auto-rebase/rebase.sh"
+RX_CP = "cp (-r )?(?P<src>\S+) (?P<dst>\S+)"
+RX_ENVSUBST = "envsubst ?< ?(?P<src>\S+) ?> ?(?P<dst>\S+)"
+
+UPDATE_MANIFESTS_START = "\nupdate_manifests() {\n"
+UPDATE_MANIFESTS_END = "\n}\n"
+
+IGNORES = [
+    "assets/embed.go",  # embeds assets into the binary
+    "assets/release/*",  # part of download_release() function
+    "assets/components/ovn/**",  # TODO: manifests are present in cluster-network-operator repository, we should start including them (what about configure-ovs.sh?)
+    "assets/components/lvms/**",  # TODO: remove when rebase automation is complete for topolvm
+    "assets/core/priority-class-openshift-user-critical.yaml",  # PriorityClass required by `oc debug` (file manually introduced from cluster-config-operator)
+    "assets/core/securityv1-local-apiservice.yaml",
+    "assets/version/microshift-version.yaml"  # empty template processed at runtime
+]
+
+DEBUG = False
+
+
+class MatchResult(Enum):
+    OK = 1
+    NOK = 2
+    IGNORED = 3
+    FAILURE = 4
+
+
+def get_asset_processing_commands():
+    """
+    Extracts contents of update_manifests() function in rebase.sh.
+    Only lines containing following commands are returned: cp, git restore, envsubst.
+    """
+    def pred(line):
+        return (line.find("cp ") != -1 or
+                line.find("git restore") != -1 or
+                line.find("envsubst <") != -1)
+
+    with open(REBASE_SCRIPT, "r") as f:
+        contents = f.read()
+        start = contents.find(UPDATE_MANIFESTS_START)
+        end = contents.find(UPDATE_MANIFESTS_END, start) + len(UPDATE_MANIFESTS_END)
+        return [line for line in contents[start:end].splitlines() if pred(line)]
```

Comment:
Is there a way for us to restructure the rebase command to make this need to parse it less complex (and possibly fragile)? For example, what if we had a separate text file of all of the files that need to be copied into the assets tree from a release payload? The rebase script could read that file, and so could this function.

Reply:
If we'd put all `cp` in the text file, file by file (no globs) - that would definitely simplify validation.
But I'm not sure if that would be so simple - there're some exceptions and quirks like clearing dirs, `git restore`-ing certain files.

We would have to tear apart current organization of the function, i.e.:
```
    #-- OpenShift control plane ---------------------------
    #-- openshift-dns -------------------------------------
    #-- openshift-router ----------------------------------
    #-- service-ca ----------------------------------------
    #-- ovn-kubernetes -----------------------------------
```
and each has following steps:
```
    # 1) Adopt resource manifests
    # 2) Render operand manifest templates like the operator would
    # 3) Make MicroShift-specific changes
    # 4) Replace MicroShift templating vars (do this last, as yq trips over Go templates)
```

If we'd run `rm`s first, then process that list with files to copy and then go back to `yq`-ing, `sed`-ing, and `git restore`-ing then this structure falls apart.
One thing that might be misleading is having to edit two files: list to copy and rebase.sh to make modifications, but other than that I think it would make the process (both adding new entries and validating) more robust.
We could go even further and move everything from `update_manifests()` to one file (json? yaml? let say a CR of some sort :D)  but not sure how time consuming would that be and would have to be tested quite hard

Reply:
OK, I didn't realize the existing logic was so complicated.

I'll push a little bit again on making it easier to parse: Would it help to have comments or other markers in the script? Something that could be an explicit interface to this code, instead of relying on code style conventions?

If you're confident of keeping the 2 things in sync as they are, we can move ahead, but I have to ask. :-)